go/vt/vtgate/planbuilder/system_tables.go

```diff
@@ -151,10 +160,63 @@ func extractInfoSchemaRoutingPredicate(in sqlparser.Expr) ([]evalengine.Expr, []
 			cmp.Right = populateValTuple(valTuples, len(colNames))
 			return sysTableSchemas, sysTableNames, nil
 		}
+	case *sqlparser.OrExpr:
+		sysTableSchemas := make([]evalengine.Expr, 0)
+		sysTableNames := make([]evalengine.Expr, 0)
+		schemas, names, err := extractInfoSchemaRoutingPredicate(cmp.Left, sysTableSchema, sysTableName)
+		if err != nil {
+			return nil, nil, err
+		}
+		sysTableSchemas = append(sysTableSchemas, schemas...)
+		sysTableNames = append(sysTableNames, names...)
+		schemas, names, err = extractInfoSchemaRoutingPredicate(cmp.Right, append(sysTableSchema, sysTableSchemas...), append(sysTableName, sysTableNames...))
+		if err != nil {
+			return nil, nil, err
+		}
+		sysTableSchemas = append(sysTableSchemas, schemas...)
+		sysTableNames = append(sysTableNames, names...)
+		return sysTableSchemas, sysTableNames, nil
+	case *sqlparser.AndExpr:
+		sysTableSchemas := make([]evalengine.Expr, 0)
+		sysTableNames := make([]evalengine.Expr, 0)
+		for _, subexpr := range sqlparser.SplitAndExpression(nil, cmp) {
+			schemas, names, err := extractInfoSchemaRoutingPredicate(subexpr, append(sysTableSchema, sysTableSchemas...), append(sysTableName, sysTableNames...))
```

Comment:
it is not a good practice to put calculation of expressions in params.

first append the list, then call the method.